Update to EFY

SubTeno opened the pull request:
- Centered the content,

- Made the video card model bigger,

- Added bigger grid card gap,

- Redesigned the subscription count 
Subscriptions page has a bug where the "Unsubscribe" text has a "-" in the end, as in "Unsubscribe -". 
Example : https://efy.piped.pages.dev/subscriptions (need to login obv)
I later discovered that the "-" represent the subscription count, and that the it mainly used as a count indicator in channel and video, so it got intertwined and i had to rewire it, if you will. 
So that also means that the " {{subscribers count}} subscribers" text should be a new localization.

- Straightened Subscription button on subscriptions page
The bug only appears after you switch to another language, like japan

- Added a bit gap between subscriber button and RSS button in channel page

- Fixed the "100% width" issue 
Example : https://efy.piped.pages.dev/channel/UCCiKUHVwRhDXpoCB8KiiYMw

- Straightened comment section. because it looks inconsistent and asymmetric

- Added a little line UI break in the middle of description div and comment div.

- Fixed video card in channel page mistakenly showing shorts video as live video
But in return, since there is no clear report of whether a video is a Shorts or not from the API, all video below 60 second is now considered as a Shorts (Channel Page only).
This is not my decision btw, as you guys were trying to do it beforehand, i was just fixing whats broken.
Example : https://efy.piped.pages.dev/channel/UCZXhZeAbY4LYynmzVe-IQ4w

- And i guess it also fixed the inconsistent and wonky thumbnail size

Update 10/10/2022

- Adjusted verified checkmark aligning (top text)

- Changed import button (in /import) element to button, that causes buggy click pointer not working

- Made the image on WatchVideo component redirects to channel

- Centered the video card relative to their parent (Useful when theres not enough space for another column in a grid but still made an extra unnecessary space, what happen is the card would justify to the left)

Update 11/10/2022

- Adjusted modal width for laptop and mobile

- Added custom instance feature (solves #1434)

Update 12/10/2022

- Resolved issues: removed :class[theme] and changed to camel case

- Added validation URL for custom instance

- Added theater feature (this along with centered margin replaces #1136)

- Fixed efy_select children (show description, show recommendation, and autoloop)  only saves localstorage to autoplay when switched

- Fixed chapter card model's width when thumbnail error
Example : https://efy.piped.pages.dev/watch?v=MZStPwNaGZ4 (segment 12 - 15 chapter has broken thumbnail)

Update 13/10/2022

- Added if statement to remove efy sidebar when embed

- Improved bottom placement mechanic of footer

Update 14/10/2022

- Fixed custom instance feature bug.

- Added v-if to remove custom instance form

- Tightened and improved URL validation custom instance even more.

- Widen content width (lessening LR margin)

Update 16/10/2022

- Replaced width to min-width in App.vue to fix handleScroll() function

- Reduced videoStep number to 30 as 100 seems unnecessary and thus loads eager content and also reduces server load, (by not loading eager image), but idk the exact number so -- yea.

- Removed custom instance modal to reduce redundancy and to activate reactivity (so that people doesn't need to reload everytime they add/remove a custom instance)

Update 17/10/2022

- Added custom instance modal back, using emit as the underlying logic. ty k'vin

dragos-efy commented:
Thanks for the effort! I will comment on the things related to the efy branch or the framework, because you changed things related to both Piped in general (which I'm not the one deciding stuff for) and the efy branch.

For comments, the app's width & the grid gap we were supposed to make them modular, so that the user can customise them from the efy sidebar, therefore I don't think that changing the defaults now would be useful, it's more of a personal choice imo and I don't wanna annoy users who prefer things the opposite way or impose my view or yours on others. But these features will indeed be implemented in the future.

I've seen a few things I liked, like the line for the description, but I haven't had time yet to look at all your changes properly. Either way, feel free to dm me on matrix if you have questions or suggestions. Thanks again for the effort and in the future try to maybe ask the devs before implementing stuff so that you don't waste your time working on something that will already be done haha

Bnyro commented on src/App.vue:
```diff
@@ -1,15 +1,15 @@
-    <div class="w-full min-h-screen px-1vw reset" :class="[theme]">
-        <NavBar />
-
+    <NavBar />
```
The Vue compiler is complaining here because `template` elements should only have one direct child. You should probably wrap all the content here into a `div` to make Vue happy :)

FireMasterK replied:
But that shouldn't be the case in Vue 3 😕

Bnyro replied:
Ouch, maybe I'm using an outdated VSCode extension version :/

Bnyro commented on src/App.vue:
```diff
@@ -1,15 +1,15 @@
-
-        <FooterComponent />
     </div>
+    <label :class="[theme]">
```
The `:class="[theme]"` is not needed here and for the `NavBar` not either, Efy doesn't use it since it handles theme differently than Piped did previously.

Bnyro commented on src/components/PreferencesPage.vue:
```diff
@@ -478,23 +483,33 @@ export default {
+        this.fetchJson("https://piped-instances.kavin.rocks/")
+            .then(resp => {
+                this.instances = resp;
+                if (this.custominstances != null)
```
I'd recommend using camel-cases here: `custominstances -> customInstances` && `cusinstance -> customInstance`

Bnyro commented:
Apart from my comments it would probably be a good idea to make the horizontal margin (/theather mode) optional/customizable in the settings since some people probably prefer to have it fill the whole width.
In the future we might be able to integrate such UI settings into efy's sidebar module, but for now I direct integration into Piped's settings is probably the best I think :)

Bnyro commented on src/components/CustomInstanceModal.vue:
```diff
@@ -0,0 +1,82 @@
+        }
+    },
+    methods: {
+        add() {
```
When the inserted URL is invalid, the app will crash after the new instance got pushed. It would probably be a good idea to validate the inserted URL and showing an `alert` when it is invalid.

SubTeno commented:
resolved

Bnyro commented on src/components/CustomInstanceModal.vue:
```diff
@@ -5,7 +5,7 @@
             <select v-model="selectedInstance">
                 <option
-                    v-for="(instance, pointer) in custominstance"
+                    v-for="(instance, pointer) in customInstances"
```
The more common term here would be index :)

Bnyro commented:
Some notes:
1. In my opinion, the theater mode should remove/reduce the margin of all the content below the video too (e.g. the comments, the description, the related videos, ...) and not only the video itself.
2. My comment concerning the margin/theater mode was meant for the landing page with the trending videos. As for know, there are big margins on the left and the right:
![2022-10-14T17:35:25,839527897+02:00](https://user-images.githubusercontent.com/82752168/195886008-84f303a3-367c-405d-a077-bb4752980250.png)
My suggestion is to add an option to remove that margin and therefore have more videos per row (same for the search results)

Apart from that, the PR should be ready to become merged soon 🎉

dragos-efy commented:
While I appreciate your effort, as I said... I still think this needs to be split into multiple PRs and a few of the things you did are supposed to be added anyway in a more modular approach.

For example in the next efy update I'm planning to add an option for custom max-width, so no need for a hard set default when it's customisable anyway. Or comments will be either as they are, or expand to the whole width and maybe even a 3rd mode, idk yet. Custom gaps can be part of an efy module as well.

I spotted some bugs / inconsistencies in my testing:

![Screenshot_20221014_232841](https://user-images.githubusercontent.com/86549690/195939021-0087c552-7e94-4070-b4b0-967f75c5bc99.png)

![Screenshot_20221014_233150](https://user-images.githubusercontent.com/86549690/195939039-df7a8a0c-868a-4f20-b62c-380b92511cbe.png)

![Screenshot_20221014_233005](https://user-images.githubusercontent.com/86549690/195939050-fb5bdef9-38d3-4454-a2dd-9347972b917a.png)

What I like tho is:

- Adjusted modal width for laptop and mobile
- Added a little line UI break in the middle of description div and comment div
- Added a bit gap between subscriber button and RSS button in channel page

I'm not sure if this is great or if making the sidebar on embeded videos fit better would be better, so I have restrained thoughts on this: `Added if statement to remove efy sidebar when embed` for now.

What I don't personally think is needed either cuz it will be done with a different approach or cuz it introduces instability (which we want to avoid so that we can make the efy branch stable and merge it with official sooner):

- Fixed chapter card model's width when thumbnail error
- Fixed efy_select children (show description, show recommendation, and autoloop) only saves localstorage to autoplay when switched (not fully sure if this one causes responsiveness issues or not tho, lmk if I'm wrong)
- Adjusted verified checkmark aligning (top text)
- Fixed inconsistent and wonky thumbnail size
- Straightened comment section. because it looks inconsistent and asymmetric
- Centered the content
- Made the video card model bigger
- Added bigger grid card gap
- Fixed the "100% width" issue
- Added if statement to remove efy sidebar when embed

My overall take is that you're trying to fix SOME things that are not a big priority now, and introduce more complexity, bugs or hard set values (rather than modularity), BUT some of the changes, especially related to just piped but not the efy branch specifically are nice additions that improve the experience, so congrats on those.

Again, I feel like these changes need to be reduced, simplified and split into more specific PRs, rather than one huge set of changes on top of a semi stable branch... beta changes on top of a beta branch is pretty unstable. This is just my take tho, but I care about people's experience a lot and I don't want them to be restrained by defaults. Thanks again for your time and dedication, my dms are still open if I can help with something.

SubTeno commented:
> While I appreciate your effort, as I said... I still think this needs to be split into multiple PRs and a few of the things you did are supposed to be added anyway in a more modular approach.
> 
> For example in the next efy update I'm planning to add an option for custom max-width, so no need for a hard set default when it's customisable anyway. Or comments will be either as they are, or expand to the whole width and maybe even a 3rd mode, idk yet. Custom gaps can be part of an efy module as well.
> 
> I spotted some bugs / inconsistencies in my testing:
> 
> ![Screenshot_20221014_232841](https://user-images.githubusercontent.com/86549690/195939021-0087c552-7e94-4070-b4b0-967f75c5bc99.png)
> 
> ![Screenshot_20221014_233150](https://user-images.githubusercontent.com/86549690/195939039-df7a8a0c-868a-4f20-b62c-380b92511cbe.png)
> 
> ![Screenshot_20221014_233005](https://user-images.githubusercontent.com/86549690/195939050-fb5bdef9-38d3-4454-a2dd-9347972b917a.png)
> 
> What I like tho is:
> 
> * Adjusted modal width for laptop and mobile
> * Added a little line UI break in the middle of description div and comment div
> * Added a bit gap between subscriber button and RSS button in channel page
> 
> I'm not sure if this is great or if making the sidebar on embeded videos fit better would be better, so I have restrained thoughts on this: `Added if statement to remove efy sidebar when embed` for now.
> 
> What I don't personally think is needed either cuz it will be done with a different approach or cuz it introduces instability (which we want to avoid so that we can make the efy branch stable and merge it with official sooner):
> 
> * Fixed chapter card model's width when thumbnail error
> * Fixed efy_select children (show description, show recommendation, and autoloop) only saves localstorage to autoplay when switched (not fully sure if this one causes responsiveness issues or not tho, lmk if I'm wrong)
> * Adjusted verified checkmark aligning (top text)
> * Fixed inconsistent and wonky thumbnail size
> * Straightened comment section. because it looks inconsistent and asymmetric
> * Centered the content
> * Made the video card model bigger
> * Added bigger grid card gap
> * Fixed the "100% width" issue
> * Added if statement to remove efy sidebar when embed
> 
> My overall take is that you're trying to fix SOME things that are not a big priority now, and introduce more complexity, bugs or hard set values (rather than modularity), BUT some of the changes, especially related to just piped but not the efy branch specifically are nice additions that improve the experience, so congrats on those.
> 
> Again, I feel like these changes need to be reduced, simplified and split into more specific PRs, rather than one huge set of changes on top of a semi stable branch... beta changes on top of a beta branch is pretty unstable. This is just my take tho, but I care about people's experience a lot and I don't want them to be restrained by defaults. Thanks again for your time and dedication, my dms are still open if I can help with something.

this is getting spicy, lets start.

> "While I appreciate your effort, as I said... I still think this needs to be split into multiple PRs"
you never mention/said anything about splitting PR , not here , not on matrix.
"and a few of the things you did are supposed to be added anyway in a more modular approach."

Alright so i think i understand now that you think me changing the default layout is useless as you keep bashing me with the same, more or less sentence -- and i think its not, first impression to me is a big role in getting people to like your product, its literally what they see first, what if they leave the site because it looks shit before they even know that option exist ?

Sure sometimes people are just comfortable in what they want their product to be, and thats fine, but what about the majority of the people that likes to be dictated by the default ? i mean if you are a chef do you really want to give your food a decent presentation and only when people critize your gonna say "we have an option too" ?

Option is called option for reason , are they not ?
Heck i would even say viewtube has a better UI/UX than piped right now, and they dont even have option (they do but only a theme option). One of the reason why i like piped is because they have good default UI/UX, not because they have option.

> "I spotted some bugs / inconsistencies in my testing"

May i know your machine information ? im sure that you're atleast on linux distro so i've tried it on ubuntu virtual machine and it has no problem whatsover. theres also a thing with tailwind where you have to wait sometimes and refresh the page before the plugin starts to kick in.


> "I'm not sure if this is great or if making the sidebar on embeded videos fit better would be better, so I have restrained thoughts on this: Added if statement to remove efy sidebar when embed for now." 

Uh why would u want to customize an embed when all the content to that is literally just a video...?

> "My overall take is that you're trying to fix SOME things that are not a big priority now, and introduce more complexity, bugs or hard set values (rather than modularity), BUT some of the changes, especially related to just piped but not the efy branch specifically are nice additions that improve the experience, so congrats on those."

Well... first of all i change things because you guys did too so i thought its okay for me to do the same things: the current branch has different layout from the stable one, you guys changed the preferences page, the gap grid, and the efy_select children (show description, show recommendation, and autoloop).. ? 

Second of all, i'm well aware i added more feature that essentially added more feature thats why i consulted to bnyro before even making a commit, heres the message i sent to him : 

"hey, i wanna ask something. i made an update to the app : added a custom instance feature (and some minor changes), the problem is i implemented it onto EFY, which makes it pretty useless -- not that i care but what i wanna ask is if its okay to do a pull request on it instead of the main branch ? im not comfortable either placing it on the main branch since its very new feature and needs more testing. "

And he literally said : 

"Yes, that's alright :)"

The theater option i added was bnyro's idea ( https://github.com/TeamPiped/Piped/pull/1545#issuecomment-1275099393 ) , so clearly i got my permission there.

And third of all, i maintain all the features i wrote (atleast until stable) so nothing to lose, right ?


Offtopic : i really dont like writing things here but as things getting hot, i feel like i have to let everyone ( and also kavin, i guess ) know

SubTeno commented:
> Some notes:
> 
> 1. In my opinion, the theater mode should remove/reduce the margin of all the content below the video too (e.g. the comments, the description, the related videos, ...) and not only the video itself.
> 2. My comment concerning the margin/theater mode was meant for the landing page with the trending videos. As for know, there are big margins on the left and the right:
>    ![2022-10-14T17:35:25,839527897+02:00](https://user-images.githubusercontent.com/82752168/195886008-84f303a3-367c-405d-a077-bb4752980250.png)
>    My suggestion is to add an option to remove that margin and therefore have more videos per row (same for the search results)
> 
> Apart from that, the PR should be ready to become merged soon 🎉

> In my opinion, the theater mode should remove/reduce the margin of all the content below the video too (e.g. the comments, the description, the related videos, ...) and not only the video itself.

Theater mode exist for the sole purpose of enlarging the video and the video only, thats it. Bringing the whole content with it is a whole-nother goofy feature that i never even seen any website does this, ever.  

> My comment concerning the margin/theater mode was meant for the landing page with the trending videos. As for know, there are big margins on the left and the right.

Widen content width. Done

> My suggestion is to add an option to remove that margin and therefore have more videos per row (same for the search results)

Dragos said he was already going to do this so this is his part.  or i could do this, but i want things to atleast be merged first before continuing as i feel like this pull request is getting too long.

Bnyro commented:
We'll probably just create a new branch with all your changes and then allow others to test these, then it can still be slightly improved before we merge it into the main efy branch.
It's probably way too late to split all these things into multiple PRs now, this would take ages and true, it would have been better, but we literally can't force you to do all your work again :)
I agree with Dragos in terms of the future modularity plans, but it won't be hard to change these things back when efy has the option/support for it then, so I don't really see an issue there at the moment.

Bnyro commented:
> Theater mode exist for the sole purpose of enlarging the video and the video only, thats it. Bringing the whole content with it is a whole-nother goofy feature that i never even seen any website does this, ever.  
> 
I'm of another opinion here. If you want to enlarge the video, you go into fullscreen. The theater mode should also enlarge the width of the content below as in https://github.com/TeamPiped/Piped/pull/1136, otherwise the feature doesn't make sense at all.

dragos-efy commented:
@SubTeno I'm not trying to heat anything up haha, as I said, I REALLY do appreciate your effort, I'm not gonna shit on it for no reason, especially cuz I'm an egoless designer, the idea wins, not who implements it. And the simple truth is that ux wise you're trying to copy what you've seen in other apps rather than actually design piped (in some regards, in others your implementation is great)... I don't care what other apps are doing, I care about providing a good experience and you don't have factual evidence for your claims in terms of what the defaults should be, because things like max-width are highly debated and divided topics in ux design. Just cuz you like them or I like them or some random apps do it, it doesn't mean it should be the default. Windows, linkedin, facebook for example are clear examples of ugly design that got dragged for far too long and stopped eventually, other apps copying them just cuz they're popular, but eventually the hype fades away and people realise that what matters is substance, not how many apps do the same thing as you.

To give you even clearer examples, the custom width and gap sizes are not space efficient and consistent in terms of variables used... that's one of the basic ux rules... you're breaking multiple of those rules, (which to be fair most people do) I can't be happy of someone proposing something that goes against consistency and ease of use, I will allow people to have an option to use the exact value in pixels or % tho, I feel like that's better, while the default is space efficient and consistent on ALL DEVICES. But I think you got that, so I won't repeat myself and save you some time there haha.

The smaller width works for text related sites more, because of reading patterns, so your take is not completely off, it works in some contexts, in fact if you go to the efy demo you'll see that I have a smaller width, so I'm not against it in general, just in this context as a default, cuz here it reduces valuable space. Searching through lots of videos is easier when you can see more at once for example. Try it on a 4k or big monitor / tv (if you can) and you'll see that up to 70% of space is wasted and the actual app area is only 30%. Unless someone has 2x scaling, which can be done from the browser or the efy sidebar, but you're proposing a default, not a default + 2x scaling (which wouldn't work on most devices anyway). See what I mean? I have actual reasons for not going with your take, I'm not just saying this to sound cool. Context matters as well, not just the idea. Your idea is great, but not in this context.

The verified checkmark is also centered to indicate grouping rather than ranking.

Which brings the next point... try your own design on multiple resolutions, browsers and devices and you'll spot the same issues from my screenshots. I use arch btw, to answer your question, but that's irrelevant cuz the piped frontend runs in a browser anyway, so I'd rather blame the browser than the distro haha. I waited patiently for everything to load, so that can't be an issue, cuz i can spot and reproduce the problems directly from dev tools, right in the css.

I also have gaps in my design, plus my time is limited on the project, I'm not perfect, and that's not the point anyway, but if I mess something up I'll acknowledge it. We're supposed to give people a great experience, not argue about who has a bigger stick or shaming others for not accepting our ideas. Design involves criticism, just cuz I criticised some of your implementations doesn't mean I criticised YOU as a person, so again I have nothing against you, you're welcome to come up with stuff, but I won't tell you that something is great if I know it's not... I'd rather disappoint you than disappoint thousands of users who use the app. For me truth matters. So I hope there's no heat perceived from my side, at least not intentionally. I'm just criticising implementations, not you as a person.

Also keep in mind that I was more vocal about what I don't like, but there's things related to piped specifically that I also like and didn't mention. Lower your guard, drink some apple juice, eat a greek salad and enjoy your day, we're not fighting here, I'm in your room, bringing the feta cheese, lemon juice and olive oil for the salad so that you can mix them up with your lettuce, tomatoes, salt and pepper 😂🥗

dragos-efy commented:
> > Theater mode exist for the sole purpose of enlarging the video and the video only, thats it. Bringing the whole content with it is a whole-nother goofy feature that i never even seen any website does this, ever.
> 
> I'm of another opinion here. If you want to enlarge the video, you go into fullscreen. The theater mode should also enlarge the width of the content below as in #1136, otherwise the feature doesn't make sense at all.

yeah, I think it makes sense only when the width is smaller, otherwise it doesn't do much, but then that's extra effort implementing 3 modes kinda... 100%, custom width and custom width + theather mode, which I guess we can do in the sidebar at some point tho, but I wouldn't call it a top priority right now

dragos-efy commented:
> We'll probably just create a new branch with all your changes and then allow others to test these, then it can still be slightly improved before we merge it into the main efy branch. It's probably way too late to split all these things into multiple PRs now, this would take ages and true, it would have been better, but we literally can't force you to do all your work again :) I agree with Dragos in terms of the future modularity plans, but it won't be hard to change these things back when efy has the option/support for it then, so I don't really see an issue there at the moment.

I think a different branch would be nice, so that when I add the custom width it will be easier for @SubTeno to adapt, without us having to change things 2 times

dragos-efy commented:
or we can merge a few things like the subscribe button, description separator and the shorts badge trick separately if we all agree with those and sorta reduce this PR by a bit so that it's easier to manage

SubTeno commented:
> > We'll probably just create a new branch with all your changes and then allow others to test these, then it can still be slightly improved before we merge it into the main efy branch. It's probably way too late to split all these things into multiple PRs now, this would take ages and true, it would have been better, but we literally can't force you to do all your work again :) I agree with Dragos in terms of the future modularity plans, but it won't be hard to change these things back when efy has the option/support for it then, so I don't really see an issue there at the moment.
> 
> I think a different branch would be nice, so that when I add the custom width it will be easier for @SubTeno to adapt, without us having to change things 2 times

I dont think we need to make a new branch, i really dont see anything wrong with merging this PR to the branch, you ,bnyro, and i  have been reviewing it for the last week, and a handy of bugs has been solved in the process, bnyro also has said "Apart from that, the PR should be ready to become merged soon 🎉". 

the only bug i've seen right now is with your archlinux compatibility, and even that is a niche bug so i dont think its a deal breaker. Not everyone can be affected by it , and i can deal with that later as the branch gets merged.

Bnyro commented on src/components/PreferencesPage.vue:
```diff
@@ -386,12 +386,55 @@
         </tbody>
     </table>
-    <CustomInstanceModal v-if="showmodal" @close="showmodal = !showmodal" />
+    <ModalComponent v-if="showmodal" @close="showmodal = !showmodal">
```
I'm not really happy with this solution: It would be better to keep the dialog as its own component and just update the custom instances list of the `PreferencesPage` when `@close` of the modal gets triggered. Just pasting all the code inside here doesn't look like a good solution here!

SubTeno replied:
Sure, but the consequent of that would be : 

1. Disabling reactivity that consequently affects user experience -- so everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not. This also defeats the purpose of single page app, and redirecting (technically it is) page is the last thing you wanna do because vue has to load everything from the beginning instead what we want is to just update a little bit part of the page, thus making it efficient, thats why SPA is good.
So not only this is inefficient, this is also irritating, i tested this myself for about 2 - 3 hours and i get irritated that i have to refresh the page, click the modal button, type it again. its frustating, trust me.

2. Data Redundancy. Remember DRY. We have to initialize 2 custominstance, making things more inefficient.

3. Unnecessary component means more workflow behind the back, we're instantiating a CustomInstanceModal, and that modal is insatantiating ModalComponent, unless its something thats used more than one case, like account login/register modal, its not worth it.


So unless you wanna kill user experience, and their hardware, i'd rather not go back.

SubTeno replied:
I also suspect you didn't understand fully the motivation behind the change and just see whats left. This is pretty much what expected when one writes long code. If theres anything that you wanna ask about the change, please dont hesitate to ask beforehand.

FireMasterK replied:
> Everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not.

You have emits in Vue, which are suited exactly for this purpose: https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events, a potential solution is to maybe add a method to add an instance in the `PreferencesPage.vue`, while the validation logic would remain in `CustomInstanceModal`.

SubTeno replied:
> > Everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not.
> 
> You have emits in Vue, which are suited exactly for this purpose: https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events, a potential solution is to maybe add a method to add an instance in the `PreferencesPage.vue`, while the validation logic would remain in `CustomInstanceModal`.

done, ty for the help.

kirkas commented:
Out of curiosity, why are we moving towards Efy as a UI framework?

The [project itself](https://efy.ooo/ui#) is still in Beta and the redesign doesn't seems to drastically change or address any UI/UX issues.

It adds a new CSS framework on top of Tailwind (Higher learning curve for new contributors) and has a few issues:

- Heavily override default browser display such as select, checkboxes and inputs (UX no-no).
- CSS: No PostCSS, purge or modularization, the [CSS is included a whole](https://github.com/dragos-efy/efy/blob/main/efy/efy.css)
- JS: no modularization, little to no comments, no post-processor. the [JS is included a whole](https://github.com/dragos-efy/efy/blob/main/efy/efy.js) (including code that affect [localStorage](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L19), includes [components by default](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L292) and tries to [add html elements to your website](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L21). 
- Has cross-browser issues, if you open https://efy.ooo/ui# on MacOS Safari, the buttons and some other elements don't display well.
- Lack of documentation and only has one contributor

On a subjective and personal level, I think the heavy usage of gradients, blurry backgrounds, rounded fonts, rounded components and jumpy animation get in the way of the overall experience, with no obvious UI/UX wins.

Bnyro commented:
> The [project itself](https://efy.ooo/ui#) is still in Beta and the redesign doesn't seems to drastically change or address any UI/UX issues.
> 
Currently https://efy.piped.pages.dev and the `efy` branch on GitHub are only there for testing, it will most likely replace the current interface as soon as it's out of beta and we reviewed the JS.
> It adds a new CSS framework on top of Tailwind (Higher learning curve for new contributors) and has a few issues:
> 
In first place it makes it easier to have a uniform design as most elements are pre-styled, second I doubt that efy has a higher learning curve than UnoCss (not Tailwind). Most likely UnoCSS will be replaced completely in the end.
> - Heavily override default browser display such as select, checkboxes and inputs (UX no-no).
>
Yes, one goal is to have it look the same, no matter which browser you're using. No idea what your issue with UX is in that terms.
> - CSS: No PostCSS, purge or modularization, the [CSS is included a whole](https://github.com/dragos-efy/efy/blob/main/efy/efy.css)
>
Yes, agree, there should be a `efy.min.css` as well as a `efy.min.js` with better loading times in the future.
> - JS: no modularization, little to no comments, no post-processor. the [JS is included a whole](https://github.com/dragos-efy/efy/blob/main/efy/efy.js) (including code that affect [localStorage](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L19), includes [components by default](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L292) and tries to [add html elements to your website](https://github.com/dragos-efy/efy/blob/main/efy/efy.js#L21). 
>
That might still need some work, agree.
> - Has cross-browser issues, if you open https://efy.ooo/ui# on MacOS Safari, the buttons and some other elements don't display well.
>
That's why it's not been going live yet, it still needs some more testing before publishing it but it's on a good way in my opinion.
> - Lack of documentation and only has one contributor
> 
Documentation is gonna be improved before the end of the beta. In fact it has more contributers than Dragos (which you don't see) as we support and communicate a lot with him via the Piped Matrix room. There were already some adjustments made to better integrate with Piped.
> On a subjective and personal level, I think the heavy usage of gradients, blurry backgrounds, rounded fonts, rounded components and jumpy animation get in the way of the overall experience, with no obvious UI/UX wins.
>
I disagree, you can change the border radius, the accent color, the background and much more yourself, if you really want to you can almost rebuild the current interface!

Cheers

RaptaG commented:
Is this on a milestone or something?
